File: C++/2021/9thOct_shagun0915.cpp
#include <iostream>
using namespace std;

int factorial(int num){
   int fact = 1, i;
   for(i=1; i<=num; i++)
   fact = fact * i;
   return fact;
}
int gcd(int x, int y) {
   if (y == 0){
       return x;
   }
   return gcd(y, x % y);
}
int main() {
   int a,b;
   cin>>a>>b;
   int x=factorial(a);
   int y=factorial(b);
   cout<<gcd(x,y);
   return 0;
}

File: Python/2021/9thOct_shagun0915.py
#python program to find GCD of factorials of any two given numbers

def factorial(num):
    fact = 1
    i=1
    while i<=num:
        fact = fact * i
        i=i+1
    return fact

def gcd(x,y):
   if y == 0:
       return x
   return gcd(y, x % y)

a = int(input())
b = int(input())
x=factorial(a)
y=factorial(b)
print(gcd(x,y))
